ReSharper.FSharp/test/src/FSharp.Tests/Parsing/ArgumentsOwnerTest.fs

```diff
@@ -0,0 +1,170 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Tests.Features
+
+open System
+open FSharp.Compiler.SourceCodeServices
+open JetBrains.ProjectModel
+open JetBrains.ReSharper.Feature.Services.ExpressionSelection
+open JetBrains.ReSharper.FeaturesTestFramework.Utils
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.DeclaredElement
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+open JetBrains.ReSharper.Plugins.FSharp.Tests.Common
+open JetBrains.ReSharper.Plugins.FSharp.Util
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.TestFramework
+open JetBrains.TextControl
+open NUnit.Framework
+
+[<AttributeUsage(AttributeTargets.Method, AllowMultiple=false)>]
+type ExpectErrorsAttribute () =
+    inherit Attribute()
+
+[<FSharpTest>]
+[<TestPackages("FSharp.Core")>]
+[<TestReferences("System.Drawing", "System", "System.Core")>]
+[<TestProjectOutputType(ProjectOutputType.CONSOLE_EXE)>]
+type ArgumentsOwnerTest() =
+    inherit BaseTestWithTextControl()
+
+    override x.RelativeTestDataPath = "parsing/arguments"
+
+    override x.DoTest(lifetime, project) =
+        let textControl = x.OpenTextControl(lifetime)
+
+        let expectErrors = x.TestMethod.GetCustomAttributes(typeof<ExpectErrorsAttribute>, true).Length > 0
+
+        let fsFile = textControl.GetFSharpFile(x.Solution)
+
+        if not expectErrors then
+            match fsFile.GetParseAndCheckResults(true, "ArgumentsOwnerTest") with
+            | None -> failwithf "GetParseAndCheckResults failed"
+            | Some results ->
+
+            let errors =
+                results.CheckResults.Errors
+                |> Array.filter (fun err -> err.Severity = FSharpErrorSeverity.Error)
+
+            match errors with
+            | [||] -> ()
+            | errors ->
+                errors
+                |> Array.map (fun err -> sprintf "(%d,%d)-(%d,%d): %s %d: %s" err.Range.StartLine err.Range.StartColumn err.Range.EndLine err.Range.EndColumn err.Subcategory err.ErrorNumber err.Message)
+                |> String.concat "\n"
+                |> failwithf "Errors occurred while type checking:\n\n%s"
+
+        let expr = FSharpTreeNodeSelectionProvider.Instance.GetExpressionInRange<IArgumentsOwner>(fsFile, textControl.Selection.OneDocumentRangeWithCaret(), true, null)
+        if isNull expr then failwithf "Failed to find IArgumentsOwner at selection"
```

Comment:
There's `JetBrains.Diagnostics.Assertion.NotNull` extension method which could be used here.